spring-boot-tools/spring-boot-loader/src/main/java/org/springframework/boot/loader/PropertiesLauncher.java

```diff
@@ -489,7 +499,7 @@ private Archive getArchive(File file) throws IOException {
 	}
 
 	private List<Archive> getNestedArchives(String root) throws Exception {
-		if (root.startsWith("/")
+		if (!root.equals("/") && root.startsWith("/")
 				|| this.parent.getUrl().equals(this.home.toURI().toURL())) {
 			// If home dir is same as parent archive, no need to add it twice.
 			return null;
```

Comment:
It's hard to write unit tests for this stuff, so I had already added new integration tests instead. However, I rebased and pushed an extra commit (2ae6a14) with some more unit tests, plus bug fixes for corner cases I found when writing those, and 2 new integration tests. None of the unit tests can get at this code path unfortunately because there is no way to change the "parent" archive for the test. I think the integration tests would fail without this line though.